src/component/Text.js

```diff
@@ -0,0 +1,162 @@
+
+      const wordsByLines = this.calculateWordsByLines(
+        this.wordsWithComputedWidth,
+        this.spaceWidth,
+        props.width
+      );
+      this.setState({ wordsByLines });
+    } else {
+      const words = props.children ? props.children.toString().split(/\s+/) : [];
+      this.setState({ wordsByLines: [{ words }] });
+    }
+  }
+
+  calculateWordWidths(props) {
+    // Calculate length of each word to be used to determine number of words per line
+    var svg = document.createElementNS('http://www.w3.org/2000/svg', 'svg');
+    var text = document.createElementNS('http://www.w3.org/2000/svg', 'text');
+    _.assign(text.style, props.style);
+    svg.appendChild(text);
+    document.body.appendChild(svg);
+
+    const words = props.children ? props.children.toString().split(/\s+/) : [];
+    const wordsWithComputedWidth = words.map(word => {
+      text.textContent = word;
+      return { word, width: text.getComputedTextLength() };
+    });
+
+    text.textContent = '\u00A0'; // Unicode space
+    const spaceWidth = text.getComputedTextLength();
+
+    document.body.removeChild(svg);
+
+    return { wordsWithComputedWidth, spaceWidth };
+  }
+
+  calculateWordsByLines(wordsWithComputedWidth, spaceWidth, lineWidth) {
+    const { fit } = this.props;
+    return wordsWithComputedWidth.reduce((result, { word, width }) => {
+      const currentLine = result[result.length - 1] || { words: [], width: 0 };
+
+      if (currentLine.words.length === 0) {
+        // First word on line
+        const newLine = { words: [word], width };
+        result.push(newLine);
+      } else if (lineWidth == null || fit || (currentLine.width + width + spaceWidth) < lineWidth) {
+        // Word can be added to an existing line
+        currentLine.words.push(word);
+        currentLine.width += width + spaceWidth;
+      } else {
+        // Word is too long to fit on existing line
+        const newLine = { words: [word], width };
+        result.push(newLine);
+      }
+
+      return result;
+    }, []);
+  }
+
+  render() {
+    const {
+      textAnchor,
+      verticalAnchor,
+      fit,
+      angle,
+      lineHeight,
+      capHeight,
+      ...textProps,
+    } = this.props;
+    const { wordsByLines } = this.state;
+    const { x, y } = textProps;
+
+    let dy;
+    switch (verticalAnchor) {
+      case 'start':
+        dy = reduceCSSCalc(`calc(${capHeight})`);
+        break;
+      case 'middle':
+        dy = reduceCSSCalc(`calc(${capHeight} / 2)`);
+        break;
+      default:
+        dy = 0;
+        break;
+    }
+
+    const transforms = [];
+    if (fit) {
+      const lineWidth = wordsByLines[0].width;
+      transforms.push(`scale(${this.props.width / lineWidth})`);
+    }
+    if (angle) {
+      transforms.push(`rotate(${angle}, ${x}, ${y})`);
+    }
+    if (transforms.length) {
+      textProps.transform = transforms.join(' ');
+    }
+
+    return (
+      <text dy={dy} textAnchor={textAnchor} {...textProps}>
```

Comment:
```jsx
const {
      ...
      className,
      ...textProps,
    } = this.props;
...


return (
      <text
        {...getPresentationAttributes(textProps)}
        className={classNames('recharts-text', className)}
        textAnchor={textAnchor}
      >
      {
        wordsByLines.map((line, index) => (
          <tspan x={x} dy={index === 0 ? startDy : lineHeight} key={index}>
            {line.words.join(' ')}
          </tspan>
        ))
      }
      </text>
    );
```

These test errors can be fixed by parsing the className of `Text`. Because `React` will check the props of native dom and svg elements.